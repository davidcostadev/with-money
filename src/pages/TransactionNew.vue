<template>
  <div>
    <md-toolbar>
      <router-link to="/">
        <md-button class="md-icon-button">
          <md-icon>keyboard_backspace</md-icon>
        </md-button>
      </router-link>
      <h3 class="md-title">Add new Transaction</h3>
    </md-toolbar>
    <transaction-form
      :transactionDate="state_month.format()"
      :onSave="onSave"
    />
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import TransactionForm from '../components/forms/Transaction';
import Transaction from '../services/transactions';
// import Accounts from '../services/accounts';

export default {
  components: {
    TransactionForm,
  },
  // data() {
  //   return {
  //     // name: '',
  //     // value: '',
  //     // AccountId: 0,
  //     // type: 'out',
  //     // isPaid: false,
  //     // transactionDate: new Date(),
  //     // accounts: [],
  //   };
  // },
  computed: {
    ...mapGetters(['state_month']),
  },
  methods: {
    async onSave(data) {
      const transaction = await Transaction.create(data);

      if (transaction.id) {
        this.$store.dispatch('showFlashMessage', 'Transanção salva com sucesso!');
      }

      this.$router.push('/');
    },
  },
  mounted() {
    // this.transactionDate = this.state_month.format();
  },
};
</script>
