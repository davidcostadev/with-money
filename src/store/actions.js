import * as types from './types';

export default {
  addUser({ commit }, payload) {
    commit(types.ADD_USER, payload);
  },
  toNextMonth({ commit }) {
    commit(types.TO_NEXT_MONTH);
  },
  toCurrentMonth({ commit }) {
    commit(types.TO_CURRENT_MONTH);
  },
  toPreviousMonth({ commit }) {
    commit(types.TO_PREVIUS_MONTH);
  },
};
