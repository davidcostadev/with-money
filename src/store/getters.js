export default {
  user(state) {
    return state.user;
  },
  current_month(state) {
    return state.current_month;
  },
  state_month(state) {
    return state.state_month;
  },
  state_month_str(state) {
    return state.state_month_str;
  },
  next_month_str(state) {
    return state.next_month_str;
  },
  previous_month_str(state) {
    return state.previous_month_str;
  },
};
