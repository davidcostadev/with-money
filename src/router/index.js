import Vue from 'vue';
import Router from 'vue-router';
import Login from '@/pages/Login';
import TransactionEdit from '@/pages/TransactionEdit';
import TransactionList from '@/pages/TransactionList';
import store from '../store';

Vue.use(Router);

const insecure = (to, from, next) => {
  const token = window.localStorage.getItem('token');

  if (!store.getters.user && !token) {
    next();
    return;
  }

  next('/');
};

const secure = (to, from, next) => {
  const token = window.localStorage.getItem('token');

  if (store.getters.user || token) {
    next();
    return;
  }

  next('/login');
};

export default new Router({
  routes: [
    {
      path: '/login',
      name: 'login',
      component: Login,
      beforeEnter: insecure,
    },
    {
      path: '/',
      name: 'TransactionList',
      component: TransactionList,
      beforeEnter: secure,
    },
    {
      path: '/transaction/:id',
      name: 'TransactionEdit',
      component: TransactionEdit,
      beforeEnter: secure,
    },
  ],
});
