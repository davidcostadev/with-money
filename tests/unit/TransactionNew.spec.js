import { createLocalVue, shallowMount } from '@vue/test-utils';
import VueMaterial from 'vue-material';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import Moment from 'moment';
import TransactionNew from '../../src/pages/TransactionNew';
import actions from '../../src/store/actions';
import getters from '../../src/store/getters';
import router from '../../src/router';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueMaterial);
localVue.use(VueRouter);

describe('TransactionNew Component', () => {
  let store;
  let nowMock;

  beforeAll(() => {
    nowMock = Date.now;
    Date.now = () => new Date('2018-09-24T00:00:00-00:00').getTime();
    store = new Vuex.Store({
      state: {
        state_month: Moment(),
      },
      getters,
      actions,
    });

    global.localStorage = {
      getItem: jest.fn().mockReturnValue(JSON.stringify({
        id: '1',
        name: 'name',
      })),
    };
  });

  afterAll(() => {
    Date.now = nowMock;
  });

  describe('Render', () => {
    it('default', () => {
      const wrapper = shallowMount(TransactionNew, { store, localVue, router });

      expect(wrapper.isVueInstance).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
