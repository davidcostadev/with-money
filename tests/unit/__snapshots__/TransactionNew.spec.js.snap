// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionNew Component Render default 1`] = `
<div>
  <mdtoolbar-stub
    mdelevation="4"
  >
    <router-link-stub
      event="click"
      tag="a"
      to="[object Object]"
    >
      <mdbutton-stub
        class="md-icon-button"
        type="button"
      >
        <mdicon-stub>
          keyboard_backspace
        </mdicon-stub>
      </mdbutton-stub>
    </router-link-stub>
     
    <h3
      class="md-title"
    >
      Add new Transaction
    </h3>
  </mdtoolbar-stub>
   
  <transactionform-stub
    accountid="0"
    categoryid="0"
    name=""
    onsave="function () { [native code] }"
    transactiondate="2018-09-23T21:00:00-03:00"
    type="in"
    value=""
  />
</div>
`;
